crates/bevy_ui/src/render/mod.rs

```diff
@@ -276,9 +276,16 @@ pub fn prepare_uinodes(
     render_queue: Res<RenderQueue>,
     mut ui_meta: ResMut<UiMeta>,
     mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    extracted_windows: Res<ExtractedWindows>,
 ) {
     ui_meta.vertices.clear();
 
+    let scale_factor = if let Some(extracted_window) = extracted_windows.get(&WindowId::primary()) {
+        extracted_window.scale_factor as f32
+    } else {
+        return;
```

Comment:
is it normal to return here, or should it be a default scale factor of 1.0?

Reply:
`bevy_ui` assumes there is one window, both for layout and rendering. It think it will need some redesign to make it work for other cases (multiple windows, render to texture, headless, etc.).
I still changed it to a default scale factor in 4a63792 to be consistent with `extract_uinodes()`.